sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/aggregator/DoubleHistogramAggregatorTest.java

```diff
@@ -0,0 +1,181 @@
+    aggregatorHandle.recordLong(10);
+
+    MetricData metricData =
+        aggregator.toMetricData(
+            Collections.singletonMap(Labels.empty(), aggregatorHandle.accumulateThenReset()),
+            0,
+            10,
+            100);
+    assertThat(metricData).isNotNull();
+    assertThat(metricData.getType()).isEqualTo(MetricDataType.HISTOGRAM);
+    assertThat(metricData.getDoubleHistogramData().getAggregationTemporality())
+        .isEqualTo(AggregationTemporality.DELTA);
+  }
+
+  @Test
+  void testHistogramCounts() {
+    assertThat(aggregator.accumulateDouble(1.1).getCounts().length)
+        .isEqualTo(boundaries.length + 1);
+    assertThat(aggregator.accumulateLong(1).getCounts().length).isEqualTo(boundaries.length + 1);
+
+    AggregatorHandle<HistogramAccumulation> aggregatorHandle = aggregator.createHandle();
+    aggregatorHandle.recordDouble(1.1);
+    HistogramAccumulation histogramAccumulation = aggregatorHandle.accumulateThenReset();
+    assertThat(histogramAccumulation).isNotNull();
+    assertThat(histogramAccumulation.getCounts().length).isEqualTo(boundaries.length + 1);
+  }
+
+  @Test
+  void testMultithreadedUpdates() throws Exception {
+    final AggregatorHandle<HistogramAccumulation> aggregatorHandle = aggregator.createHandle();
+    final Histogram summarizer = new Histogram();
+    int numberOfThreads = 10;
+    final long[] updates = new long[] {1, 2, 3, 5, 7, 11, 13, 17, 19, 23};
+    final int numberOfUpdates = 1000;
+    final CountDownLatch startingGun = new CountDownLatch(numberOfThreads);
+    List<Thread> workers = new ArrayList<>();
+    for (int i = 0; i < numberOfThreads; i++) {
+      final int index = i;
+      Thread t =
+          new Thread(
+              () -> {
+                long update = updates[index];
+                try {
+                  startingGun.await();
+                } catch (InterruptedException e) {
+                  throw new RuntimeException(e);
+                }
+                for (int j = 0; j < numberOfUpdates; j++) {
+                  aggregatorHandle.recordLong(update);
+                  if (ThreadLocalRandom.current().nextInt(10) == 0) {
+                    summarizer.process(aggregatorHandle.accumulateThenReset());
+                  }
+                }
+              });
+      workers.add(t);
+      t.start();
+    }
+    for (int i = 0; i <= numberOfThreads; i++) {
+      startingGun.countDown();
+    }
+
+    for (Thread worker : workers) {
+      worker.join();
+    }
+    // make sure everything gets merged when all the aggregation is done.
+    summarizer.process(aggregatorHandle.accumulateThenReset());
+
+    assertThat(summarizer.accumulation)
+        .isEqualTo(HistogramAccumulation.create(101000, new long[] {5000, 5000, 0, 0}));
+  }
+
+  private static final class Histogram {
+    private final ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
```

Comment:
Simple mutex here too (we're not taking a readlock anywhere so have no advantage of using this special lock)